src/main.c

```diff
@@ -118,7 +118,7 @@ int main(void) {
 	LCD_disp_str((uint8_t*)buf, len, 0, 64 - 6, FONT6X6);
 	printf("\nRunning on an %s", buf);
 
-	len = snprintf(buf, sizeof(buf), "%s", Version_GetGitVersion());
```

Comment:
This introduces a potential buyer overflow for future devs, try sizeof(buf) - 1

Reply:
Oh wow, that was fast 

As far as I have understood the original problem, buf is not the
problem. The problem is that the LCD_disp_str() function also uses fixed
length buffers according to display pixel size and these buffers will
overflow if you try to print texts resulting in more pixels than the
display width. At a 6x6 font the maximum is 21 characters, that's why I
limited to it.

Does that explain it a bit?

Reply:
In that case, I would suggest introducing a common define used between the output code and buffer allocation,  which makes the relationship explicit